buildenv/jenkins/containerbuild.groovy

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+def ARCH = (params.ARCH) ? params.ARCH : "unknown"
+def DOCKER_FILE = (params.DOCKER_FILE) ? params.DOCKER_FILE : "unknown"
+def IMAGE_NAME = (params.IMAGE_NAME) ? params.IMAGE_NAME : "unknown"
+def DOCKER_URL = (params.DOCKER_URL) ? params.DOCKER_URL : "https://hub.docker.com/r"
+
+def NODE = (params.NODE) ? params.NODE : "sw.tool.docker && hw.arch.${ARCH}"
+def NAMESPACE = (params.NAMESPACE) ? params.NAMESPACE : "eclipseopenj9"
+def FOLDER = "/omr"
+def REPOSITORY = "${NAMESPACE}${FOLDER}/jenkins-agent-${IMAGE_NAME}"
+
+timeout(time: 5, unit: 'HOURS') {
+    timestamps {
+        node(NODE) {
+            try{
+                def TEMP_DESC = (currentBuild.description) ? currentBuild.description + "<br>" : ""
+                currentBuild.description = TEMP_DESC + "<a href=${JENKINS_URL}/computer/${NODE_NAME}>${NODE_NAME}</a><br>"
+                currentBuild.description += "Docker image:<a href=${DOCKER_URL}/${REPOSITORY}>${REPOSITORY}</a>"
+                stage("Build") {
+                    sh "rm -rf docker-workspace-${IMAGE_NAME}"
+                    dir("docker-workspace-${IMAGE_NAME}") {
+                        sh "curl -L ${DOCKER_FILE} -o Dockerfile-${IMAGE_NAME}"
+                        sh "docker build -f ./Dockerfile-${IMAGE_NAME} -t ${IMAGE_NAME} ."
+                    }
+                }
+                stage("Push") {
+                    def CREDENTIALS_ID = (params.credentialsId) ? params.credentialsId : 'docker-login'
```

Comment:
These can be shortened to
`def CREDENTIALS_ID = (params.credentialsId) ?: 'docker-login'`

Reply:
Updated.